### PR TITLE
Adds SPI driver for Tiva

### PR DESCRIPTION
**Pull request by balazsracz**

### CODE REVIEW

File: src/freertos_drivers/ti/TivaSPI.hxx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+        switch (bitsPerWord)
+        {
+            default:
+            case 8:
+                return transfer_polled<uint8_t>(msg);
+            case 16:
+                return transfer_polled<uint16_t>(msg);
+            case 32:
+                return transfer_polled<uint32_t>(msg);
```
Comment (bakerstu) [inline src/freertos_drivers/ti/TivaSPI.hxx:239]: 32-bit transfers not supported by hardware, should we have an assert here?

File: src/freertos_drivers/ti/TivaSPI.hxx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    {
+        // Disable port first for reconfiguration.
+        HWREG(base_ + SSI_O_CR1) &= ~SSI_CR1_SSE;
+        
+        HWREG(base_ + SSI_O_CR0) = spiCr0_;
+        HWREG(base_ + SSI_O_CPSR) = spiPrescaler_;
+        HWREG(base_ + SSI_O_CR1) = spiCr1_;
+    }
+
+    OSSem *sem_; /**< reference to the semaphore belonging to this bus */
```
Comment (bakerstu) [inline src/freertos_drivers/ti/TivaSPI.hxx:305]: Does this get used anywhere? It may be dead code. Should probably check the CC3220 SPI driver too.

**Review by bakerstu — APPROVED** [review]

File: src/freertos_drivers/ti/TivaSPI.hxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/ti/TivaSPI.hxx:305]: I believe it is used to signal the DMA complete interrupt.

Comment (balazsracz) [reply src/freertos_drivers/ti/TivaSPI.hxx:305]: Which explains why it seems unused, because DMA is still a TODO.

**Pull request merged**